src/solana/solanaEngine.ts

```diff
@@ -382,18 +388,19 @@ export class SolanaEngine extends CurrencyEngine<SolanaTools> {
     return edgeTransaction
   }
 
-  async signTx(edgeTransaction: EdgeTransaction): Promise<EdgeTransaction> {
+  async signTx(
+    edgeTransaction: EdgeTransaction,
+    privateKeys: JsonObject
+  ): Promise<EdgeTransaction> {
+    const solPrivateKeys = asSolPrivateKeys(this.currencyInfo.pluginId)(
+      privateKeys
+    )
     const { unsignedSerializedSolTx } = getOtherParams(edgeTransaction)
     if (unsignedSerializedSolTx == null)
       throw new Error('Missing unsignedSerializedSolTx')
 
     const keypair = Keypair.fromSecretKey(
-      Uint8Array.from(
-        Buffer.from(
-          this.walletInfo.keys[`${this.currencyInfo.pluginId}Key`],
-          'hex'
-        )
-      )
+      Uint8Array.from(Buffer.from(solPrivateKeys.privateKey, 'hex'))
```

Comment:
A `Buffer` is a `Uint8Array` already, so no need to convert. But also, we should just use `base16.parse` here too.